RevenueCatUI/CustomerCenter/Views/OpenAppStoreViewController.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  OpenAppStoreViewController.swift
+//
+//  Created by JayShortway on 19/08/2024.
+
+#if CUSTOMER_CENTER_ENABLED
+
+import Foundation
+import StoreKit
+import UIKit
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+class OpenAppStoreViewController: UIViewController, SKStoreProductViewControllerDelegate {
+    let onDismiss: (SKStoreProductViewController) -> Void
+
+    init(
+        onDismiss: @escaping (SKStoreProductViewController) -> Void = { viewController in
+            viewController.dismiss(animated: true)
+        }
+    ) {
+        self.onDismiss = onDismiss
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    /// Opens the App Store for the provided productId in-app using SKStoreProductViewController. Falls back to
+    /// redirecting to the App Store if this fails.
+    func openAppStore(productId: UInt) {
```

Comment:
The [documentation](https://developer.apple.com/documentation/storekit/skstoreproductparameteritunesitemidentifier) says that the `productId`, or _iTunes item identifier_, is an instance of `NSNumber`. I opted to take a `UInt` as input, so we only accept positive integers, as that is the only type of ID I've seen in the wild. Please let me know if this is not true. 

The conformity to `NSNumber` leads me to assume that IDs cannot begin with a `0`, as I think `NSNumber` would deem that `0` insignificant and strip it. Please also correct me if I'm wrong.